### PR TITLE
add skill for awk usage

### PR DESCRIPTION
**Pull request by guazhang**
If your PR is related to a contribution to the taxonomy, please, fill
out the following questionnaire. If not, replace this whole text and the
following questionnaire with whatever information is applicable to your PR.


**Describe the contribution to the taxonomy**

   add a skill for awk usage

<!-- A concise description of what the contribution brings, replace "..." in the bullet list -->

- Returns a more accurate usage about awk tool
- Show how to use specific parameters

    
**Response from the original model**

<!-- Response from the original model -->

```
>>> How to use awk to deal `for` loop?                                                                                                                                                                                            [S][default]
INFO 2024-04-07 08:00:25,806 _client.py:1026 HTTP Request: POST http://127.0.0.1:8000/v1/chat/completions "HTTP/1.1 200 OK"
╭─────────────────────────────────────────────────────────────────────────────────────────────────────────── merlinite-7b-Q4_K_M ────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ I'm sorry for the confusion earlier. In `awk`, you don't need to use a `for` loop to process each line in a file, as `awk` does that for you. It is designed to process each line of input as a record, so you don't need to explicitly    │
│ loop over the lines.                                                                                                                                                                                                                       │
│                                                                                                                                                                                                                                            │
│ Here's an example:                                                                                                                                                                                                                         │
│                                                                                                                                                                                                                                            │
│ ```bash                                                                                                                                                                                                                                    │
│ awk '{print $1}' file.txt                                                                                                                                                                                                                  │
│ ```                                                                                                                                                                                                                                        │
│                                                                                                                                                                                                                                            │
│ This command will print out the first field from each line in the file. If you want to perform different actions based on conditions, you can use `if` statements in `awk`. For example:                                                   │
│                                                                                                                                                                                                                                            │
│ ```bash                                                                                                                                                                                                                                    │
│ awk '{if ($1 > 10) print $1, "is greater than 10"}' file.txt                                                                                                                                                                               │
│ ```                                                                                                                                                                                                                                        │
│                                                                                                                                                                                                                                            │
│ This command will print out any lines where the first field is greater than 10, along with a message indicating that it is greater than 10.                                                                                                │
│                                                                                                                                                                                                                                            │
│ If you want to loop through lines in a file and print out specific lines that meet certain conditions, you can use `grep` with a regex pattern:                                                                                            │
│                                                                                                                                                                                                                                            │
│ ```bash                                                                                                                                                                                                                                    │
│ grep '^field1' file.txt                                                                                                                                                                                                                    │
│ ```                                                                                                                                                                                                                                        │
│                                                                                                                                                                                                                                            │
│ This command will print out any lines that start with `field1`. You can adjust the pattern as needed to match specific lines in the file.                                                                                                  │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── elapsed 48.146 seconds ─╯
>>>                                                                                                                                                                                                                               
```

**lab generate results**
<!-- What you received from the original model in response to your input, 
replace "..." -->

```
  Q> What is the most common value in the third field?
I> 
A> Calculate the most common value in the third field:

awk '{c[$3]++} END {for (i in c) if (c[i] > max) {max = c[i]; val = i}} 1; {print val, max}' file.txt


Q> 4. How to find lines where the third field contains only digits?
I> 
A> Find lines where the third field contains only digits:

awk '$3 ~ /^[0-9]+$/ {print}' file.txt

```



**Response from the fine-tuned model**


<!-- Generate a synthetic dataset based on your newly added seed data; train the model 
with the synthetic data and now re-test the model's response with the same prompt.
Replace "..." with what you receive with the finetuned model. -->

```
  blocked by laptop performance issues
```

**Contribution checklist**

<!-- Insert an x between the empty brackets: [ ] >> [x] -->

- [x] The contribution was tested with `lab generate`
- [x] No errors or warnings were produced by `lab generate`
- [x] All [commits are signed off](https://github.com/instruct-lab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [x] The `qna.yaml` file contains at least 5 `seed_examples`
- [x] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)

### CODE REVIEW

**Review by linl-rh — APPROVED** [review]
Looks good to me.
